cpp/thirdparty/versions.txt

```diff
@@ -42,7 +42,7 @@ ARROW_JEMALLOC_BUILD_VERSION=5.2.1
 ARROW_LZ4_BUILD_VERSION=v1.9.3
 # mimalloc 1.6.7 didn't build on Visual Studio 2015
 # https://github.com/microsoft/mimalloc/issues/353
-ARROW_MIMALLOC_BUILD_VERSION=v1.6.4
+ARROW_MIMALLOC_BUILD_VERSION=v1.7.2
```

Comment:
Need to check that it won't cause any performance regressions.

Reply:
@lidavidm how can I verify that https://github.com/apache/arrow/pull/10024 doesn't apply here?

Reply:
As far as I can see that should only apply to 2.0.0 and 1.7.x should be safe (the commenter in https://github.com/microsoft/mimalloc/issues/363 claims 1.7.0 doesn't affect them at least)

Reply:
And the problematic test is still there - so assuming tests pass we should be OK

Reply:
Thanks!